[Lite interpreter] Pass shared_ptr properly

iseeyuan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#33667 [Lite interpreter] Pass shared_ptr properly**

Pass shared_ptr properly according to C++ guidances. Thank @kimishpatel for pointing it out.

Differential Revision: [D20111001](https://our.internmc.facebook.com/intern/diff/D20111001)

Commit:
[Lite interpreter] Pass shared_ptr by reference

Commit:
Update on "[Lite interpreter] Pass shared_ptr by reference"

Commit:
Update on "[Lite interpreter] Pass shared_ptr by reference"

kimishpatel commented on torch/csrc/jit/mobile/interpreter.cpp:
```diff
@@ -13,7 +13,7 @@ namespace jit{
 std::ostream& operator<<(std::ostream& out, Instruction inst);
 namespace mobile {
-InterpreterState::InterpreterState(std::shared_ptr<Code> code) : code_(code) {
+InterpreterState::InterpreterState(std::shared_ptr<Code> code) : code_(std::move(code)) {
```
Should we make this constructor take rvalue? `std::shared_ptr<Code>&& code`? That way the caller understands that the value maybe moved? Although all of this is self-contained, making it explicit might be better?

iseeyuan replied:
As discussed offline, rvalue ref cannot be used here because the ownership of code may be shared out of InterpreterState.

kimishpatel reviewed (changes requested):
Left a comment.

kimishpatel reviewed (approved):
LGTM

Commit:
Update on "[Lite interpreter] Pass shared_ptr properly"


Pass shared_ptr properly according to C++ guidances. Thank @kimishpatel for pointing it out.

Differential Revision: [D20111001](https://our.internmc.facebook.com/intern/diff/D20111001)
